src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package ruslan.shastkiv.bookstore.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ruslan.shastkiv.bookstore.dto.order.OrderDto;
+import ruslan.shastkiv.bookstore.dto.order.OrderItemDto;
+import ruslan.shastkiv.bookstore.dto.order.PlaceOrderRequestDto;
+import ruslan.shastkiv.bookstore.dto.order.UpdateOrderStatusRequestDto;
+import ruslan.shastkiv.bookstore.exception.EmptyShoppingCartException;
+import ruslan.shastkiv.bookstore.exception.EntityNotFoundException;
+import ruslan.shastkiv.bookstore.mapper.OrderItemMapper;
+import ruslan.shastkiv.bookstore.mapper.OrderMapper;
+import ruslan.shastkiv.bookstore.model.Order;
+import ruslan.shastkiv.bookstore.model.OrderItem;
+import ruslan.shastkiv.bookstore.model.ShoppingCart;
+import ruslan.shastkiv.bookstore.model.User;
+import ruslan.shastkiv.bookstore.repository.order.OrderItemRepository;
+import ruslan.shastkiv.bookstore.repository.order.OrderRepository;
+import ruslan.shastkiv.bookstore.service.cart.ShoppingCartService;
+import ruslan.shastkiv.bookstore.service.user.UserService;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final UserService userService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrderByUserId(Authentication authentication,
+                                       PlaceOrderRequestDto requestDto) {
+        User user = userService.getUser(authentication);
```

Comment:
why do you need `getUser` method if you have `getUserId` method
here you are using userId
`shoppingCartService.findShoppingCart(user.getId());`

Reply:
realy)
I think that these are echoes from the very first versions of the implementation of this method